src/apy/apy.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ApolloError } from 'apollo-server-express';
+import { isAuth } from 'src/utils/isAuth';
+import { catchResult } from 'src/utils/withError';
+import { Repository } from 'typeorm';
+import { APY } from './entities/APY.entity';
+
+@Injectable()
+export class APYService {
+  constructor(@InjectRepository(APY) private apyRepository: Repository<APY>) {}
+
+  async getCurrentAPY(token: string) {
+    try {
+      await isAuth();
+      const [apy, apyErr] = await catchResult(
+        this.apyRepository.find({
+          where: {
+            token: token,
+          },
+          skip: 0,
+          take: 1,
+          order: { date: 'DESC' },
+        })
+      );
+      if (!apy[0]) return `APY for ${token} is not available right now. Please try again later.`;
```

Comment:
Aquí mejor regresen el error porfa, porque en el front no podemos hacer conditional rendering. No podemos saber si hay error porque regresa el string directamente

```suggestion
      if (!apy[0]) return throw new ApolloError('APY for ${token} is not available right now. Please try again later');
```